sdk/core/core-lro/src/poller.ts

```diff
@@ -0,0 +1,148 @@
+import { PollOperation } from "./pollOperation";
+
+export type CancelOnProgress = () => void;
+
+export interface PollProgressSubscriber<TProperties, TResult> {
+  id: string;
+  conditional: (op?: PollOperation<TProperties, TResult>) => boolean;
+  callback: (op?: PollOperation<TProperties, TResult>) => void;
+}
+
+export abstract class Poller<TProperties, TResult> {
+  private stopped: boolean;
+  private resolve?: (value?: TResult) => void;
+  private reject?: (error: Error) => void;
+  private pollOncePromise?: Promise<void>;
+  private cancelPromise?: Promise<PollOperation<TProperties, TResult>>;
+  private promise: Promise<TResult>;
+  private pollProgressSubscribers: PollProgressSubscriber<TProperties, TResult>[] = [];
+  public operation: PollOperation<TProperties, TResult>;
+
+  constructor(operation: PollOperation<TProperties, TResult>, stopped: boolean = false) {
+    this.operation = operation;
+    this.stopped = stopped;
+    this.promise = new Promise(
+      (
+        resolve: (result?: TResult) => void,
+        reject: (error: Error) => void
+      ) => {
+        this.resolve = resolve;
+        this.reject = reject;
+      }
+    );
+    this.operation.state.started = true;
```

Comment:
should we only consider ourselves started after the first request is sent?

Reply:
ok!